botocore/compat.py

```diff
@@ -148,7 +148,7 @@ def total_seconds(delta):
 # Checks to see if md5 is available on this system. A given system might not
 # have access to it for various reasons, such as FIPS mode being enabled.
 try:
-    hashlib.md5()
+    hashlib.md5(usedforsecurity=False)
```

Comment:
Recording why each of these is used to justify that it's not security related, but I have no concerns with any of them.  First one: 

This one is used for checking if the MD5 algorithm is used in the customer's version of hashlib; this is sometimes removed in FIPS compliant environments, so we use this check just to set the `MD5_AVAILABLE` attribute in compat.py.